java.lang.reflect.InvocationTargetException

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
IntelliJ 2020.2.1

### Operating System 
macOS

### TeXiFy IDEA version
0.7
### Description
I saw that TeXiFy had produced the following exception. I'm not sure what I did to produce it. Maybe the stacktrace is enough for you to know what happened? Otherwise I can try to reproduce it.


### Stacktrace
```
java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor107.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:246)
	at com.sun.proxy.$Proxy21.onTextAvailable(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:220)
	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:153)
	at com.intellij.util.io.BaseOutputReader.sendText(BaseOutputReader.java:206)
	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:190)
	at com.intellij.util.io.BaseOutputReader.readAvailableNonBlocking(BaseOutputReader.java:102)
	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:67)
	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:160)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:207)
	at com.intellij.util.io.BaseDataReader.lambda$start$0(BaseDataReader.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: nl.hannahsten.texifyidea.TeXception: Extra closing parenthesis: could not close a file which was not opened. Please report the log output to the issue tracker.
	at nl.hannahsten.texifyidea.run.latex.logtab.LatexFileStack.update(LatexFileStack.kt:98)
	at nl.hannahsten.texifyidea.run.latex.logtab.LatexOutputListener.processNewTextLatex(LatexOutputListener.kt:112)
	at nl.hannahsten.texifyidea.run.latex.logtab.LatexOutputListener.processNewText(LatexOutputListener.kt:86)
	at nl.hannahsten.texifyidea.run.latex.logtab.LatexOutputListener.onTextAvailable(LatexOutputListener.kt:76)
	... 23 more
```

Thanks! With that version I don't get the above mentioned exception anymore!

I do get the following error, but that might be unrelated.


<details>
  <summary>Exception. Click to expand</summary>
  
  ```
com.intellij.diagnostic.PluginException: Could not initialize class nl.hannahsten.texifyidea.intentions.LatexToggleSmartQuotesIntention [Plugin: nl.rubensten.texifyidea]
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateExtensionWithPicoContainerOnlyIfNeeded(ComponentManagerImpl.kt:681)
	at com.intellij.serviceContainer.LazyExtensionInstance.createInstance(LazyExtensionInstance.java:43)
	at com.intellij.serviceContainer.LazyExtensionInstance.getInstance(LazyExtensionInstance.java:36)
	at com.intellij.serviceContainer.BaseKeyedLazyInstance.getInstance(BaseKeyedLazyInstance.java:37)
	at com.intellij.codeInsight.intention.impl.config.IntentionActionWrapper.getDelegate(IntentionActionWrapper.java:102)
	at com.intellij.codeInsight.intention.impl.config.IntentionActionWrapper.getFamilyName(IntentionActionWrapper.java:53)
	at com.intellij.codeInsight.intention.impl.config.IntentionActionWrapper.getFullFamilyName(IntentionActionWrapper.java:89)
	at com.intellij.codeInsight.intention.impl.config.IntentionManagerSettings.getFamilyName(IntentionManagerSettings.java:163)
	at com.intellij.codeInsight.intention.impl.config.IntentionManagerSettings.isEnabled(IntentionManagerSettings.java:176)
	at com.intellij.codeInsight.intention.impl.config.IntentionManagerImpl.getAvailableIntentions(IntentionManagerImpl.java:208)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.getActionsToShow(ShowIntentionsPass.java:316)
	at com.intellij.codeInsight.daemon.impl.ShowIntentionsPass.doCollectInformation(ShowIntentionsPass.java:226)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:399)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1110)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:391)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:367)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:170)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:365)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:181)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class nl.hannahsten.texifyidea.intentions.LatexToggleSmartQuotesIntention
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateExtensionWithPicoContainerOnlyIfNeeded(ComponentManagerImpl.kt:678)
	... 29 more
  ```
</details>

--- a/build.gradle
+++ b/build.gradle
@@ -22,7 +22,7 @@ plugins {
 }
 
 group 'nl.hannahsten'
-version '0.7'
+version '0.7.1-alpha.1.5'
 
 repositories {
     mavenCentral()

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
@@ -4,6 +4,8 @@ import com.intellij.openapi.diagnostic.Logger
 import nl.hannahsten.texifyidea.TeXception
 import nl.hannahsten.texifyidea.run.latex.logtab.LatexLogMagicRegex.LINE_WIDTH
 import nl.hannahsten.texifyidea.run.latex.logtab.LatexLogMagicRegex.lineNumber
+import nl.hannahsten.texifyidea.util.containsAny
+import nl.hannahsten.texifyidea.util.firstIndexOfAny
 import nl.hannahsten.texifyidea.util.remove
 import java.util.ArrayDeque
 
@@ -15,7 +17,10 @@ class LatexFileStack(
 
     private var shouldSkipNextLine = false
 
-    private val debug = false
+    // Usage: set to true to output all file openings/closings to idea.log
+    // Then open the LaTeX log and the relevant part of idea.log in IntelliJ, and start from the idea.log parenthesis which
+    // was closed incorrectly, use the LaTeX log and brace matching to find which file should have been closed, then check where it was actually closed, etc.
+    private val debug = true
     private val logger = Logger.getInstance("LatexFileStack")
 
     init {
@@ -26,6 +31,12 @@ class LatexFileStack(
     private var currentCollectingFile = ""
 
     override fun peek(): String? = if (isEmpty()) null else super.peek()
+
+    fun pushFile(file: String, line: String) {
+        push(file)
+        if (debug) logger.info("$line ---- Opening $file")
+        if (file.containsAny(setOf("(", ")"))) throw TeXception("$file is probably not a valid file")
+    }
 
     /**
      * AFTER all messages have been filtered, look for file openings and closings.
@@ -49,14 +60,14 @@ class LatexFileStack(
         }
 
         if (currentCollectingFile.isNotEmpty()) {
-            // Files may end halfway the line
-            val endIndex = if (")" in line) line.indexOf(')') else line.length
+            // Files may end halfway the line, or right at the end of a line
+            // Assume there are no spaces in the path (how can we know where the file ends?)
+            val endIndex = if (setOf(" ", ")", "(").any { it in line }) line.firstIndexOfAny(' ', ')', '(') else line.length
             currentCollectingFile += line.substring(0, endIndex).remove("\n")
             // Check if this was the last part of the file
             if (line.substring(0, endIndex).length < LINE_WIDTH) {
                 // Assume that paths can be quoted, but there are no " in folder/file names
-                push(currentCollectingFile.trim('"'))
-                if (debug) logger.info("$line ---- Opening ${currentCollectingFile.trim('"')}")
+                pushFile(currentCollectingFile.trim('"'), line)
                 currentCollectingFile = ""
             }
         }
@@ -88,9 +99,7 @@ class LatexFileStack(
                     currentCollectingFile += file
                 }
                 else {
-                    push(file)
-                    if (debug) logger.info("$line ---- Opening $file")
-
+                    pushFile(file, line)
                 }
             }
 

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
@@ -20,7 +20,7 @@ class LatexFileStack(
     // Usage: set to true to output all file openings/closings to idea.log
     // Then open the LaTeX log and the relevant part of idea.log in IntelliJ, and start from the idea.log parenthesis which
     // was closed incorrectly, use the LaTeX log and brace matching to find which file should have been closed, then check where it was actually closed, etc.
-    private val debug = true
+    private val debug = false
     private val logger = Logger.getInstance("LatexFileStack")
 
     init {

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
@@ -116,9 +116,9 @@ class LatexFileStack(
                         throw TeXception("Extra closing parenthesis: could not close a file which was not opened. Please report the log output to the issue tracker. Line content: $line")
                     }
                     pop()
-
                 }
             }
+
             linePart = linePart.substring(result.range.last + 1)
             result = fileRegex.find(linePart)
         }

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexFileStack.kt
@@ -1,5 +1,6 @@
 package nl.hannahsten.texifyidea.run.latex.logtab
 
+import com.intellij.openapi.diagnostic.Logger
 import nl.hannahsten.texifyidea.TeXception
 import nl.hannahsten.texifyidea.run.latex.logtab.LatexLogMagicRegex.LINE_WIDTH
 import nl.hannahsten.texifyidea.run.latex.logtab.LatexLogMagicRegex.lineNumber
@@ -13,6 +14,9 @@ class LatexFileStack(
 ) : ArrayDeque<String>() {
 
     private var shouldSkipNextLine = false
+
+    private val debug = false
+    private val logger = Logger.getInstance("LatexFileStack")
 
     init {
         addAll(file)
@@ -52,6 +56,7 @@ class LatexFileStack(
             if (line.substring(0, endIndex).length < LINE_WIDTH) {
                 // Assume that paths can be quoted, but there are no " in folder/file names
                 push(currentCollectingFile.trim('"'))
+                if (debug) logger.info("$line ---- Opening ${currentCollectingFile.trim('"')}")
                 currentCollectingFile = ""
             }
         }
@@ -78,11 +83,14 @@ class LatexFileStack(
                 // Check if file spans multiple lines
                 // +1 because the starting ( is not in the group
                 // -1 because the newline is not here
-                if (file.length + 1 >= LINE_WIDTH - 1 || (line.length >= LINE_WIDTH && line.trim().endsWith(file))) {
+                // Files could span exactly the full line width, but not continue on the next line
+                if (!file.endsWith(".tex") && (file.length + 1 >= LINE_WIDTH - 1 || (line.length >= LINE_WIDTH && line.trim().endsWith(file)))) {
                     currentCollectingFile += file
                 }
                 else {
                     push(file)
+                    if (debug) logger.info("$line ---- Opening $file")
+
                 }
             }
 
@@ -94,10 +102,12 @@ class LatexFileStack(
                 }
                 if (notClosedNonFileOpenParentheses > 0) notClosedNonFileOpenParentheses--
                 else {
+                    if (debug) logger.info("$line ---- Closing ${peek()}")
                     if (isEmpty()) {
                         throw TeXception("Extra closing parenthesis: could not close a file which was not opened. Please report the log output to the issue tracker. Line content: $line")
                     }
                     pop()
+
                 }
             }
             linePart = linePart.substring(result.range.last + 1)

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexLogMagicRegex.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexLogMagicRegex.kt
@@ -15,7 +15,7 @@ object LatexLogMagicRegex {
     const val LATEX_ERROR_REGEX: String = "!" // error
     const val PDFTEX_ERROR_REGEX: String = "!pdfTeX error:"
     const val LATEX_WARNING_REGEX: String = "LaTeX( Font)? Warning:" // warning
-    const val PACKAGE_REGEX: String = """(?<package>[\d\w]+)""" // package error/warning?
+    const val PACKAGE_REGEX: String = """(?<package>[\d\w-\.]+)""" // package error/warning?
     const val REFERENCE_REGEX: String = """(?<label>(`|').+')""" // reference warning
     const val PACKAGE_WARNING_CONTINUATION = "\\(\\w+\\) {${"Package warning:".length}}"
     const val DUPLICATE_WHITESPACE = """\s{2,}"""
@@ -27,7 +27,7 @@ object LatexLogMagicRegex {
      */
 
     /** A variation on [FILE_LINE_REGEX] by lualatex (?) */
-    val directLuaError = """^\((?!\s*\.\.\.)(.+\.\w+)\)(\[.+\])?:(?<line>\d+): (?<message>.*)""".toRegex()
+    val directLuaError = """^\((?!\s*\.\.\.)(.+\.\w+)\)(\[.+])?:(?<line>\d+): (?<message>.*)""".toRegex()
     val fixMeError = """FiXme (Fatal )?Error: '(?<message>.+)' on input line (?<line>\d+).""".toRegex()
 
     /*

--- a/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexLogMessageExtractor.kt
+++ b/src/nl/hannahsten/texifyidea/run/latex/logtab/LatexLogMessageExtractor.kt
@@ -76,15 +76,17 @@ object LatexLogMessageExtractor {
         // Assumes
         if (TEX_MISC_WARNINGS.any { text.removeSuffix(newText).startsWith(it) }) {
             var messageText = if (LatexOutputListener.isLineEndOfMessage(newText, text)) text.remove(newText) else text
+
+            // Don't include the second line if it is not part of the message
+            // (Do this before cleaning up the messageText)
+            if (LatexOutputListener.isLineEndOfMessage(newText, text.remove(newText))) {
+                messageText = messageText.remove(newText).trim()
+            }
+
             messageText = messageText.remove("LaTeX Warning: ")
                 // Improves readability, and at the moment we don't have an example where this would be incorrect
                 .trim('(', ')', '[', ']')
                 .replace(DUPLICATE_WHITESPACE.toRegex(), " ")
-
-            // Don't include the second line if it is not part of the message
-            if (LatexOutputListener.isLineEndOfMessage(newText, text.remove(newText))) {
-                messageText = messageText.remove(newText).trim()
-            }
 
             return LatexLogMessage(
                 messageText,

--- a/src/nl/hannahsten/texifyidea/util/Strings.kt
+++ b/src/nl/hannahsten/texifyidea/util/Strings.kt
@@ -214,6 +214,17 @@ fun String.runCommand(): String? {
     }
 }
 
+/**
+ * Index of first occurrence of any of the given chars. Return last index if chars do not appear in the string.
+ */
+fun String.firstIndexOfAny(vararg chars: Char): Int {
+    var index = length - 1;
+    for (char in chars) {
+        if (indexOf(char) != -1) index = min(index, indexOf(char))
+    }
+    return index;
+}
+
 /** If this contains any of the given set. */
 fun CharSequence.containsAny(set: Set<String>) = set.any { this.contains(it) }
 

--- a/src/nl/hannahsten/texifyidea/util/Strings.kt
+++ b/src/nl/hannahsten/texifyidea/util/Strings.kt
@@ -218,11 +218,11 @@ fun String.runCommand(): String? {
  * Index of first occurrence of any of the given chars. Return last index if chars do not appear in the string.
  */
 fun String.firstIndexOfAny(vararg chars: Char): Int {
-    var index = length - 1;
+    var index = length - 1
     for (char in chars) {
         if (indexOf(char) != -1) index = min(index, indexOf(char))
     }
-    return index;
+    return index
 }
 
 /** If this contains any of the given set. */

--- a/test/nl/hannahsten/texifyidea/run/LatexLogFileFinderTest.kt
+++ b/test/nl/hannahsten/texifyidea/run/LatexLogFileFinderTest.kt
@@ -124,20 +124,55 @@ class LatexLogFileFinderTest : BasePlatformTestCase() {
 
     fun testSingleFile() {
         val log = """
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
-thfunctions.base.code.tex")
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
-thfunctions.round.code.tex")
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
-thfunctions.misc.code.tex")
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
+            thfunctions.base.code.tex")
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
+            thfunctions.round.code.tex")
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
+            thfunctions.misc.code.tex")
+            
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/latex/pgf/math\pgfmath
+            .sty"
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/utilities\
+            pgfkeys.code.tex"
+            ("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
+            thfloat.code.tex")))
+            """.trimIndent()
+        var stack = LatexFileStack()
+        log.split('\n').forEach {
+            stack = stack.update(it + "\n")
+        }
+        assertTrue(stack.isEmpty())
+    }
 
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/latex/pgf/math\pgfmath
-.sty"
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/utilities\
-pgfkeys.code.tex"
-("C:\Users\thoscho\AppData\Local\Programs\MiKTeX 2.9\tex/generic/pgf/math\pgfma
-thfloat.code.tex")))
-""".trimIndent()
+    fun testFileEndingHalfwayOpen() {
+        val log = """
+            (/usr/local/texlive/2020/texmf-dist/tex/latex/glossaries/base/glossaries.sty
+            (/usr/local/texlive/2020/texmf-dist/tex/latex/glossaries/styles/glossary-super.
+            sty (/usr/local/texlive/2020/texmf-dist/tex/latex/supertabular/supertabular.sty
+            ))
+            (/usr/local/texlive/2020/texmf-dist/tex/latex/glossaries/styles/glossary-tree.s
+            ty))
+        """.trimIndent()
+        var stack = LatexFileStack()
+        log.split('\n').forEach {
+            stack = stack.update(it + "\n")
+        }
+        assertTrue(stack.isEmpty())
+    }
+
+    fun testFileEndingHalfwayOpen2() {
+        val log = """
+(/usr/local/texlive/2020/texmf-dist/tex/latex/pgf/basiclayer/pgfcore.sty
+(/usr/local/texlive/2020/texmf-dist/tex/latex/pgf/systemlayer/pgfsys.sty
+(/usr/local/texlive/2020/texmf-dist/tex/generic/pgf/systemlayer/pgfsys.code.tex
+(/usr/local/texlive/2020/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex
+(/usr/local/texlive/2020/texmf-dist/tex/generic/pgf/utilities/pgfkeysfiltered.c
+ode.tex))
+(/usr/local/texlive/2020/texmf-dist/tex/generic/pgf/systemlayer/pgf.cfg)
+(/usr/local/texlive/2020/texmf-dist/tex/generic/pgf/systemlayer/pgfsys-pdftex.d
+ef) )))
+        """.trimIndent()
         var stack = LatexFileStack()
         log.split('\n').forEach {
             stack = stack.update(it + "\n")

--- a/test/nl/hannahsten/texifyidea/run/LatexOutputListenerTest.kt
+++ b/test/nl/hannahsten/texifyidea/run/LatexOutputListenerTest.kt
@@ -656,4 +656,45 @@ class LatexOutputListenerTest : BasePlatformTestCase() {
 
         testLog(log, expectedMessages)
     }
+
+    fun `test no file`() {
+        val log = """
+            (./main.tex
+            (/home/thomas/texlive/2020/texmf-dist/tex/latex/base/size10.clo)
+(/home/thomas/texlive/2020/texmf-dist/tex/latex/l3backend/l3backend-pdfmode.def
+)
+No file synctest.aux.
+[1{/home/thomas/texlive/2020/texmf-var/fonts/map/pdftex/updmap/pdftex.map}]
+[2] [3] (/home/thomas/GitRepos/random-tex/out/synctest.aux) )</home/thomas/texl
+ive/2020/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb></home/thomas/texl
+        """.trimIndent()
+
+        val expectedMessages = setOf(
+            LatexLogMessage("No file synctest.aux.", "./main.tex", -1, WARNING)
+        )
+
+        testLog(log, expectedMessages)
+    }
+
+    fun `test glossaries-extra warning`() {
+        val log = """
+            (/home/thomas/texlive/2020/texmf-dist/tex/latex/glossaries/styles/glossary-tree
+            .sty)
+            (/home/thomas/texlive/2020/texmf-dist/tex/latex/glossaries-extra/glossaries-ext
+            ra-bib2gls.sty)
+            
+            Package glossaries-extra Warning: No file `glossaries-option4-bib2gls.glstex' o
+            n input line 17.
+            
+            
+            (/home/thomas/texlive/2020/texmf-dist/tex/latex/l3backend/l3backend-pdfmode.def
+            ) (/home/thomas/GitRepos/random-tex/out/glossaries-option4-bib2gls.aux)
+        """.trimIndent()
+
+        val expectedMessages = setOf(
+                LatexLogMessage("glossaries-extra: No file `glossaries-option4-bib2gls.glstex'", null, 17, WARNING)
+        )
+
+        testLog(log, expectedMessages)
+    }
 }

